library/tasks/Attach10xBarcodes.wdl

```diff
@@ -32,22 +34,45 @@ task Attach10xBarcodes {
   command {
     set -e
 
-    Attach10xBarcodes \
-      --r1 "${r1_fastq}" \
-      ${"--i1 " + i1_fastq} \
-      --u2 "${r2_unmapped_bam}" \
-      --output-bamfile barcoded.bam \
-      --whitelist "${whitelist}"
+    if [ "${chemistry}" == "tenX_v2" ]
```

Comment:
It might be cleaner to have different tasks for the different chemistries and run them using an `if` condition at the workflow level instead of the task level. That would mean a task always produces the "same" types of outputs and it's clear what was run just by looking at a timing diagram rather than having to drill into the metadata to get inputs at runtime.

Reply:
Would normally agree but eventually we only want one command 'AttachBarcodes' with chemistry dependent parameters (that are ideally read from a file not from he command line). We are keeping Attach10xBarcodes because we don't want to alter the code path for V2 and risk breaking things. The whole thing needs to be revised anyway because its inefficient and currently the slowest pipeline step.

Reply:
ah ok

Reply:
@kbergin tagging you to be aware for future prioritization